internal/storage/bucket_handle.go

```diff
@@ -96,3 +96,39 @@ func (b *bucketHandle) StatObject(ctx context.Context, req *gcs.StatObjectReques
 
 	return
 }
+
+func (bh *bucketHandle) CreateObject(ctx context.Context, req *gcs.CreateObjectRequest) (o *gcs.Object, err error) {
+	obj := bh.bucket.Object(req.Name)
+
+	if req.GenerationPrecondition != nil && *req.GenerationPrecondition != 0 {
+		if req.MetaGenerationPrecondition != nil && *req.MetaGenerationPrecondition != 0 {
+			obj = obj.If(storage.Conditions{GenerationMatch: *req.GenerationPrecondition, MetagenerationMatch: *req.MetaGenerationPrecondition})
+		} else {
+			obj = obj.If(storage.Conditions{GenerationMatch: *req.GenerationPrecondition})
+		}
+	}
+
+	// Creating a NewWriter with requested attributes, using Go Storage Client.
+	// Chuck size for resumable upload is default i.e. 16MB.
+	wc := obj.NewWriter(ctx)
+	wc.ChunkSize = 0 // This enables one shot upload.
```

Comment:
This change disables the chunking mechanism, which improves the performance and the memory usage during the upload. But it prevents the writer in retrying in case of a transient error.

@vadlakondaswetha Please let me know, what should we do here?
There are three options:-
(a) We can make it configurable - need to pass an argument for this.
(b) We can with the current change - no retry but improves memory and performance.
(c) The default chunk size - 16 MB.

Reply:
Configurable shouldn't be required for now.
how it improves the memory usage in case of single shot upload?
I would suggest create a bug to load test the write flow to check what should be the chunk size

Reply:
When chunk size is zero, it doesn't keep buffer in order to retry the failed request.

Reply:
If ChunkSize is set to zero, chunking will be disabled and the object will be uploaded in a single request without the use of a buffer.

Reply:
Sure. I'll create a bug to tune the value of chunk-size according to the performance trade-off.

Reply:
Cool.